pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -659,29 +661,74 @@ type InterfaceTarget struct {
 }
 
 type Alias struct {
+	name        string
+	userDefined bool
+}
+
+// Package private, responsible to interact with xml.Encoder and xml.Decoder
+type userAliasXML struct {
 	Name string `xml:"name,attr"`
 }
 
-type UserAlias Alias
+// Package private, responsible to interact with json.Marshal and json.Unmarshal
+type userAliasJSON struct {
```

Comment:
Can we just have one internal struct, something like:

```
type userAliasMarshal struct {
	Name string `xml:"name,attr"`
	UserDefined bool `xml:"-"`
}
```

The `xml:"-"` annotation makes the xml marshaller ignore this field.

Reply:
Great tip. Thanks. code changed following the suggestion.